File: examples/space_weather/file_sw.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Initialize Space Weather Providers from files
"""

import brahe as bh

# Method 1: Default Provider -> Uses packaged data file within Brahe
sw_file_default = bh.FileSpaceWeatherProvider.from_default_file()
bh.set_global_space_weather_provider(sw_file_default)

# Method 2: Custom File Path -> Replace with actual file path
if False:  # Change to True to enable custom file example
    sw_file_custom = bh.FileSpaceWeatherProvider.from_file(
        "/path/to/sw19571001.txt",  # Replace with actual file path
        "Hold",  # Extrapolation: "Zero", "Hold", or "Error"
    )
    bh.set_global_space_weather_provider(sw_file_custom)

File: examples/space_weather/file_sw.rs
//! Initialize Space Weather Providers from files

use brahe as bh;
use bh::space_weather::SpaceWeatherExtrapolation;
use std::path::Path;

fn main() {
    // Method 1: Default Provider -> Uses packaged data file within Brahe
    let sw_file_default =
        bh::space_weather::FileSpaceWeatherProvider::from_default_file().unwrap();
    bh::space_weather::set_global_space_weather_provider(sw_file_default);

    // Method 2: Custom File Path -> Replace with actual file path
    if false {
        // Change to true to enable custom file example
        let sw_file_custom = bh::space_weather::FileSpaceWeatherProvider::from_file(
            Path::new("/path/to/sw19571001.txt"), // Replace with actual file path
            SpaceWeatherExtrapolation::Hold,
        )
        .unwrap();
        bh::space_weather::set_global_space_weather_provider(sw_file_custom);
    }
}
